### ISSUE TITLE
No error on identical instance ID in static consumer group mode

### ISSUE THREAD

**Issue by Iadgarov** [twmb/franz-go#479]
Hi, 

Perhaps I'm missing the way to check for such an error, but I cant seem to cause a failure when creating two client, in the same consumer group, with the same instance ID.
According to the Kafka spec we should be getting a FencedInstanceID error, as defined in `kerr`. ([KIP-345](https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances))

I have run this against Kafka 2.5.1 and can confirm that the broker logs do in fact mentioned returning this error

```
[2023-06-24 15:09:33,711] ERROR given member.id myistance-dac57156-ae34-4b3c-8f90-98d6f4b0aa6d is identified as a known static member myistance, but not matching the expected member.id myistance-02a09bac-1a4a-453d-b5f9-6e124ea65b77 during join-group, will respond with instance fenced error (kafka.coordinator.group.GroupMetadata)
```

The way I tested it was simply creating two client like so:

```
opts := []kgo.Opt{
		kgo.SeedBrokers(container.Address(kafka.BrokerPort)),
		kgo.ConsumerGroup("mygroup"),
		kgo.InstanceID("myistance"),
		kgo.ConsumeTopics("topic-1"),
		kgo.DisableAutoCommit(),
	}

	_, err = kgo.NewClient(opts...)
	if err != nil {
		die("unable to create client: %v", err)
	}

	_, err = kgo.NewClient(opts...)
	if err != nil {
		die("unable to create client: %v", err)
	}
```
When trying with confluent kafka go I seem to get an error, although it appears to be in the form of a `panic` admittedly.. 

Any assistance is much appreciated.

**Comment by twmb** [comment]
This was a bug, good find. This is fixed in v1.13.6.

**Comment by twmb** [comment]
Notably: you should have been seeing this error from polling, and you will now.
